service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersController.java

```diff
@@ -87,8 +93,23 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody Call
 
             directions.forEach((key, value) -> caseDetails.getData().put(key.getValue(), value));
 
-            caseDetails.getData()
-                .put("judgeAndLegalAdvisor", caseData.getStandardDirectionOrder().getJudgeAndLegalAdvisor());
+            caseDetails.getData().put(JUDGE_AND_LEGAL_ADIVSOR_KEY,
+                caseData.getStandardDirectionOrder().getJudgeAndLegalAdvisor());
+        }
+
+        if (isNotEmpty(caseData.getAllocatedJudge())) {
+            JudgeAndLegalAdvisor judgeAndLegalAdvisor;
+
+            if (isNotEmpty(caseData.getStandardDirectionOrder())
+                && isNotEmpty(caseData.getStandardDirectionOrder().getJudgeAndLegalAdvisor())) {
+                judgeAndLegalAdvisor = caseData.getStandardDirectionOrder().getJudgeAndLegalAdvisor();
+            } else {
+                judgeAndLegalAdvisor = JudgeAndLegalAdvisor.builder().build();
+            }
```

Comment:
Is here the best place for this logic? perhaps we could move this to a private method that returns the judge and legal advisor? what do you think?